wolfcrypt/src/random.c

```diff
@@ -983,6 +983,29 @@ static void Entropy_StopThread(void)
     }
 }
     /* end if defined(HAVE_PTHREAD) */
+
+#elif defined(_WIN32) /* USE_WINDOWS_API */
+
+/* Get the high resolution time counter.
+ *
+ * @return  64-bit time that is the nanoseconds of current time.
+ */
+static WC_INLINE word64 Entropy_TimeHiRes(void)
+{
+    static int           init = 0;
+    static LARGE_INTEGER freq;
+    LARGE_INTEGER        count;
+
+    if (!init) {
+        QueryPerformanceFrequency(&freq);
+        init = 1;
+    }
+
+    QueryPerformanceCounter(&count);
+
+    return (word64)(count.QuadPart / (freq.QuadPart / 1000 / 1000));
```

Comment:
The comment indicates nanoseconds, but the returned value is in microseconds. 

divide by another `/ 1000`

or I think you can use
`(word64)(count.QuadPart / freq.QuadPart * 1000000000);`